common/metrics/src/lib.rs

```diff
@@ -76,6 +77,19 @@ use std::{
     thread, time,
 };
 
+const NUM_METRICS_NAME: &str = "num_metrics";
+pub static NUM_METRICS: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!(
+        NUM_METRICS_NAME,
+        "Number of metrics in certain states",
+        &["state"]
+    )
+    .unwrap()
+});
+
+const MAX_COUNTERS: usize = 1000;
```

Comment:
let's make this configurable, e.g., in the config/logging / metrics?

Reply:
Sure